projects/ngx-maplibre-gl/src/lib/map/map.component.ts

```diff
@@ -262,10 +279,13 @@ export class MapComponent
 
   ngOnDestroy() {
     this.mapService.destroyMap();
+    for (const subscription of this.subscriptions) {
+      subscription.unsubscribe();
+    }
   }
 
   async ngOnChanges(changes: SimpleChanges) {
-    await this.mapService.mapCreated$.toPromise();
+    await firstValueFrom(this.mapService.mapCreated$);
```

Comment:
@HarelM this change is breaking the unit tests at the moment.

Reply:
Triggering `mapCreated$` with a value fixed the tests

Reply:
Great, sorry, it was using a deprecated method...